Fix rendermime-interfaces

I just created an extension for jupyterlab v3 from the cookiecutter. I tried to install it before changing anything, and it failed.
```
yarn run v1.21.1
$ jlpm build:lib && jlpm build:labextension:dev
$ tsc --sourceMap
node_modules/@jupyterlab/rendermime/lib/widgets.d.ts:8:31 - error TS2420: Class 'RenderedCommon' incorrectly implements interface 'IRenderer'.
  The types of 'title.owner.layout' are incompatible between these types.
    Type 'import("/home/carlos/Documents/blockly-widgets/jupyterlab_blockly_widgets/node_modules/@lumino/widgets/types/layout").Layout | null' is not assignable to type 'import("/home/carlos/Documents/blockly-widgets/jupyterlab_blockly_widgets/node_modules/@jupyterlab/rendermime-interfaces/node_modules/@lumino/widgets/types/layout").Layout | null'.
      Type 'import("/home/carlos/Documents/blockly-widgets/jupyterlab_blockly_widgets/node_modules/@lumino/widgets/types/layout").Layout' is not assignable to type 'import("/home/carlos/Documents/blockly-widgets/jupyterlab_blockly_widgets/node_modules/@jupyterlab/rendermime-interfaces/node_modules/@lumino/widgets/types/layout").Layout'.
        Property '[Symbol.iterator]' is missing in type 'import("/home/carlos/Documents/blockly-widgets/jupyterlab_blockly_widgets/node_modules/@lumino/widgets/types/layout").Layout' but required in type 'import("/home/carlos/Documents/blockly-widgets/jupyterlab_blockly_widgets/node_modules/@jupyterlab/rendermime-interfaces/node_modules/@lumino/widgets/types/layout").Layout'.

8 export declare abstract class RenderedCommon extends Widget implements IRenderMime.IRenderer {
                                ~~~~~~~~~~~~~~

  node_modules/@jupyterlab/rendermime-interfaces/node_modules/@lumino/widgets/types/layout.d.ts:83:14
    83     abstract [Symbol.iterator](): IterableIterator<Widget>;
                    ~~~~~~~~~~~~~~~~~
    '[Symbol.iterator]' is declared here.
```

The problem is that we updated the dependencies of `@jupyterlab/rendermime-interfaces` to lumino v2 and released it as a minor version (v3.8.0). When creating a new extension for jupyterlab v3 it installs  `@jupyterlab/rendermime-interfaces` v3.8.0 with lumino v2.

We should bump the version of the packages to 4 or maybe allow both versions of Lumino.

_Originally posted by @hbcarlos in https://github.com/jupyterlab/jupyterlab/issues/14335#issuecomment-1553340328_

--- a/buildutils/src/ensure-package.ts
+++ b/buildutils/src/ensure-package.ts
@@ -80,9 +80,19 @@ export async function ensurePackage(
       seenDeps[name] = await getDependency(name);
     }
     if (deps[name] !== seenDeps[name]) {
-      messages.push(`Updated dependency: ${name}@${seenDeps[name]}`);
+      const inRange =
+        deps[name].includes('||') &&
+        deps[name]
+          .split(/\|\|/)
+          .map(v => v.trim())
+          .includes(seenDeps[name]);
+
+      if (!inRange) {
+        messages.push(`Updated dependency: ${name}@${seenDeps[name]}`);
+
+        deps[name] = seenDeps[name];
+      }
     }
-    deps[name] = seenDeps[name];
   });
 
   await Promise.all(promises);

--- a/buildutils/src/ensure-package.ts
+++ b/buildutils/src/ensure-package.ts
@@ -80,14 +80,14 @@ export async function ensurePackage(
       seenDeps[name] = await getDependency(name);
     }
     if (deps[name] !== seenDeps[name]) {
-      const inRange =
+      const oneOf =
         deps[name].includes('||') &&
         deps[name]
           .split(/\|\|/)
           .map(v => v.trim())
           .includes(seenDeps[name]);
 
-      if (!inRange) {
+      if (!oneOf) {
         messages.push(`Updated dependency: ${name}@${seenDeps[name]}`);
 
         deps[name] = seenDeps[name];

--- a/packages/rendermime-interfaces/package.json
+++ b/packages/rendermime-interfaces/package.json
@@ -31,8 +31,8 @@
     "watch": "tsc -b --watch"
   },
   "dependencies": {
-    "@lumino/coreutils": "^2.1.1",
-    "@lumino/widgets": "^2.1.1"
+    "@lumino/coreutils": "^1.11.0 || ^2.1.1",
+    "@lumino/widgets": "^1.37.2 || ^2.1.1"
   },
   "devDependencies": {
     "rimraf": "~3.0.0",

